addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/handler/GatewayHandler.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lightify.handler;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.lightify.internal.discovery.LightifyDeviceDiscoveryService;
+import org.openhab.binding.lightify.internal.link.LightifyLink;
+import org.osgi.framework.ServiceRegistration;
+
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ADDRESS;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_GATEWAY;
+
+/**
+ * The {@link org.eclipse.smarthome.core.thing.binding.BridgeHandler} implementation to handle commands
+ * and status of the OSRAM Lightify gateway device. This handler uses a native implementation of the
+ * Lightify proprietary protocol to communicate with the Lightify gateway, therefore no internet connection
+ * is necessary to utilize the OSRAM public API.
+ *
+ * @author Christoph Engelbert (@noctarius2k) - Initial contribution
+ */
+public class GatewayHandler extends BaseBridgeHandler {
+
+    /**
+     * Supported {@link ThingTypeUID}s for this handler
+     */
+    public static final Set<ThingTypeUID> SUPPORTED_TYPES = Collections.singleton(THING_TYPE_LIGHTIFY_GATEWAY);
+
+    private LightifyDeviceDiscoveryService discoveryService;
+    private ServiceRegistration<?> serviceRegistration;
+
+    private String address;
+    private LightifyLink lightifyLink;
+    private ScheduledFuture<?> futureConnect;
+
+    public GatewayHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+    }
+
+    @Override
+    public void initialize() {
+        address = getConfig().get(PROPERTY_ADDRESS).toString();
+        scheduleConnect(true);
```

Comment:
I just verified this finding using an Osram Lightify Gateway and a RGBW bulb. The thing status for the gateway and also for the bulb stay "ONLINE", no matter if the gateway is even connected or the bulb has power.

Additionally: I am wondering if the protocol allows for retrieving the actual status of the devices so it can be shown in the openHAB gui. For example, if I switch on a bulb in openHAB and then use the Osram app to switch it off, openHAB still shows the bulb as "on". It would be nice if the switch in openHAB would represent the "off state" of the bulb.